### ISSUE TITLE
Bug with specific prices

### ISSUE THREAD

**Issue by dngraphisme** [PrestaShop/PrestaShop#35463]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

There is a problem with the specific prices. Example :

1. We create a specific price on a variation of a product
2. We create a specific price on all variations of the product, with a minimum of products to purchase: 2
3. in the front office the discounted price does not change if you take 2 quantities. He remains stuck on the specific price of point n°1

<img width="1648" alt="doesnt-work" src="https://github.com/PrestaShop/PrestaShop/assets/5530905/c1819960-bbdb-4b76-8036-6fb56ebf4f19">

----

Note: Works if you do not put all the variations, but with only one variation of the product

<img width="1641" alt="work" src="https://github.com/PrestaShop/PrestaShop/assets/5530905/260d91c9-88dc-4aa1-bc1f-f4b5cb2a30e3">

### Steps to reproduce

1. We create a specific price on a variation of a product
2. We create a specific price on all variations of the product, with a minimum of products to purchase: 2
3. in the front office the discounted price does not change if you take 2 quantities. He remains stuck on the specific price of point n°1

https://github.com/PrestaShop/PrestaShop/assets/5530905/ffc9c7a1-5d4d-4c5b-aa81-31aea7784de2

### PrestaShop version(s) where the bug happened

1.7.8.10

### Your company or customer's name goes here (if applicable).

Kosmo

**Comment by ChillCode** [comment]
Hi @dngraphisme 

Go to Shop Parameters->Product configuration and check discounts per product not per combination and try again.

**Comment by dngraphisme** [comment]
Hello @ChillCode thanks for your feedback.

I had already thought about it and it was already configured like that.

No it doesn't work on 1.7.8.7. Same problem on version 8.1.4


https://github.com/PrestaShop/PrestaShop/assets/5530905/6a2c3598-a4c2-44b4-bc9d-1e292ea00b53

**Comment by florine2623** [comment]
Hello @dngraphisme ,

I can't reproduce your issue.

https://github.com/PrestaShop/PrestaShop/assets/16019289/587a7acd-9e6f-4d8b-88db-6f44096e997e

Am I missing something ?

Waiting for your feedback.
Thanks!

**Comment by dngraphisme** [comment]
Hello @florine2623,

Please reproduce what I do in the back office on the product :


1. We create a specific price on a variation of a product
2. We create a specific price on all variations of the product, with a minimum of products to purchase: 2
3. in the front office the discounted price does not change if you take 2 quantities. He remains stuck on the specific price of point n°1

----

In your video at no time does quantity 2 on the product discount the product.

You only have the discount on a quantity that is triggered.

Please watch my video carefully, it's explicit.

Thanks :)

**Comment by florine2623** [comment]
Hello @dngraphisme ,

The issue is that the volume discount table is always displayed no matter the combination you are choosing ?

My config :
<img width="1187" alt="Screenshot 2024-03-08 at 15 25 28" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/d67b90ce-00dc-4640-93ac-76bc8cb68f29">

My 2 combination have the same table discounts, even thought it should only be displayed for the color Black :
<img width="1117" alt="Screenshot 2024-03-08 at 15 25 34" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/061575d5-1650-4df6-ba91-eb9837b60d00">
<img width="1123" alt="Screenshot 2024-03-08 at 15 25 39" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/9a1bc992-6dd6-4b8c-9580-756c2fc1fd7f">

**Comment by dngraphisme** [comment]
Hello @florine2623 ,

As already explained previously :

1. Create a reduction on a variation
2. Create a reduction on ALL VARIATIONS with for example a greater discount

<img width="1378" alt="Capture d’écran 2024-03-11 à 07 57 12" src="https://github.com/PrestaShop/PrestaShop/assets/5530905/9c9af499-a2c4-4be6-8d75-3731e148fbd2">


3. On the front, if you choose the variation already discounted, Prestashop does not take the strongest discount (namely point no. 2)


https://github.com/PrestaShop/PrestaShop/assets/5530905/3043545f-5f8a-4991-970b-4c3906cbd39a

Please check my attachment. Thanks

**Comment by ChillCode** [comment]
Hi,

As i understand what PS is doing is taking in consideration the specific price for combination and ignoring the all combinations one?

I think PS will always do that sine you're giving a combination, but not 100% sure which should be the behavior i usually only add one specific price.

**Comment by dngraphisme** [comment]
@ChillCode Exactly, rather than taking the discount with a minimum of 2 products (which is higher: 12€) it changes nothing and remains on the discount of the variation where the discount is lower. Which is not logical.... The goal of buying in bulk and getting a better discount on the product...

**Comment by ChillCode** [comment]
You can also add the price per each combination, so then you really get control of each specific price.

What I usually do is create a specific price per each combination, so if there are 50.000 combinations, and have 10 range of discounts like 1, 10, 25, 50, 100, 200, 400, 500, 1000, 2000, the table will end up with 500.000 specific prices, so then the merchant can control every price separately and apply other discounts etc...

Maybe @kpodemski or @florine2623 can shed light on this.

**Comment by dngraphisme** [comment]
@ChillCode Yes that's what I did but it's still not normal :)

**Comment by kpodemski** [comment]
If the quantity discount is on the product, the setting @ChillCode mentioned should be set to "Products", but my understanding is that it still doesn't solve the issue, and it doesn't detect the discount for flat products. It indeed sounds like a bug.

**Comment by dngraphisme** [comment]
Hello @florine2623, any news or update for this please ?

**Comment by dngraphisme** [comment]
Hello @florine2623 , any news or update for this please ?

**Comment by kpodemski** [comment]
@dngraphisme if I understand correctly @ChillCode provided you with a workaround for this problem, right? let me know

also, even if we get this problem as reproduced, the chances it will be fixed some time soon are very low

**Comment by dngraphisme** [comment]
Hello @kpodemski  No absolutely not, I have already explained everything to you above. What we provide @ChillCode  is just a workaround and not a resolution. Thanks....
